tests/simulator_test.cpp

```diff
@@ -2,82 +2,160 @@
 #include "main/simulator.h"
 #include "gtest/gtest.h"
 
-// ! OLD ENERGY COLLECTORS TEST
-/*
-TEST_F(SceneManagerTest, EnergyCollectorPositionsCheck) {
-  for (const auto &collectorPtr : manager.getEnergyCollectors()) {
-    std::cout << *collectorPtr << std::endl;
+using constants::kPi;
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+
+const int kSkipNumCollectors = 37;
+const float kSkipFrequency = 1000;
+const Vec3 kVecZero(0, 0, 0);
+
+class FakeModel : public AbstractModel {
+
+public:
+  explicit FakeModel(bool empty) : empty_(empty){};
+  std::vector<objects::TriangleObj *> triangles() const override { return {}; }
+  float height() const override { return 0; }
+  float sideSize() const override { return 0; }
+  bool empty() const { return empty_; }
+
+private:
+  bool empty_;
+};
+
+class EnergyCollectorTest : public ::testing::Test {
+public:
+  EnergyCollectorTest() : nonEmptyModel(false), emptyModel(true){};
+  enum class HitResult { ENERGY_COLLECTORS_EMPTY, NO_HIT, HIT };
+
+protected:
+  std::vector<std::unique_ptr<objects::EnergyCollector>> energyCollectors;
+  FakeModel nonEmptyModel, emptyModel;
```

Comment:
Yeah, this way it can be more readable. I need to figure out how to sue empty method, when I have empty_ member in the class, because now its not doing anything - so I leave constructor as its is right now